receiver/sqlserverreceiver/factory.go

```diff
@@ -43,10 +43,15 @@ func setupQueries(cfg *Config) []string {
 		queries = append(queries, getSQLServerDatabaseIOQuery(cfg.InstanceName))
 	}
 
-	if cfg.MetricsBuilderConfig.Metrics.SqlserverResourcePoolDiskThrottledReadRate.Enabled ||
```

Comment:
This issue could pretty easily happen again. It may be wise to change this functionality to always scrape and record the metrics, even if none are enabled. This would be some level of performance hit in some situations, but reduce complexity and reduce risk of potentially hard to find bugs in the future.